### ISSUE TITLE
Implementation status

### ISSUE THREAD

**Issue by chicoxyzzy** [tc39/proposal-flatMap#71]
- [x] - V8 / Chrome 69 
- [x] - SpiderMonkey / Firefox 62
- [x] - JSC / Safari 12
- [x] - ChakraCore / Edge 19 probably

**Comment by keithamus** [comment]
ChakraCore issue https://github.com/Microsoft/ChakraCore/issues/5543

**Comment by chicoxyzzy** [comment]
ChakraCore PR https://github.com/Microsoft/ChakraCore/pull/5573

**Comment by chicoxyzzy** [comment]
@bterlson @michaelficarra all major browser has implemented these features. test262 already have tests ([flat](
https://github.com/tc39/test262/tree/master/test/built-ins/Array/prototype/flat), [flatMap](
https://github.com/tc39/test262/tree/master/test/built-ins/Array/prototype/flatMap)). It seems that proposal needs just PR to ecma262 to become ES2019

**Comment by ljharb** [comment]
It’d also need committee consensus.

**Comment by chicoxyzzy** [comment]
It could be too late for consensus when most of browsers will start to ship it on by default.

- Chrome 69 will be released on September 4
- Firefox 62 will be released on September 5
- Safari 12 will be released on September 12 presumably

While next TC39 meeting will happen on September 25

**Comment by ljharb** [comment]
I’m not sure what you mean by “too late”; stage 4 requires 1-2 shipping implementations minimum (at which point it’s web incompatible to unship it), but there’s no maximum.

**Comment by chicoxyzzy** [comment]
Oh. I though that by consensus you mean something like approvement from committee. That unshipping compatibility rule makes sense to me.

**Comment by ljharb** [comment]
Yes, the committee needs to officially approve stage 4 as well.

**Comment by ljharb** [comment]
I think you do; that something can’t be removed isn’t the same thing as that thing being in the spec, because that requires consensus. It’s all in the process document.

**Comment by tschneidereit** [comment]
This is correct in principle, as the TC39 process requires explicit committee approval for stage advancement. In practice, this is a rubber stamp and an opportunity to do a round of applause in a committee meeting :)

**Comment by michaelficarra** [comment]
Championing a proposal is (most of the time) a whole lot of work, and that round of applause means a lot. Looking forward to finally putting a bow on this one.
